api.go

```diff
@@ -51,21 +51,21 @@ func CreatePod(podConfig PodConfig) (*Pod, error) {
 	}
 
 	// Initialize the network.
-	err = p.network.init(&(p.config.NetworkConfig))
+	netNsPath, netNsCreated, err := p.network.init(p.config.NetworkConfig)
 	if err != nil {
 		return nil, err
 	}
 
 	// Execute prestart hooks inside netns
-	err = p.network.run(p.config.NetworkConfig.NetNSPath, func() error {
+	err = p.network.run(netNsPath, func() error {
```

Comment:
Well I agree it does not need to be executed from the netns, but we are passing a PID, and this PID has to be in the expected netns so that Docker can deduce the netns from this PID. And because we're passing the PID of our running thread, we need this thread to be in the netns when we're calling into the hook.